src/flymake-collection-define.el

```diff
@@ -84,7 +84,7 @@ doesn't exist: %s" dir))
      (let ((temporary-file-directory ,temp-dir)
            (basename (file-name-nondirectory (or (buffer-file-name)
                                                  (buffer-name)))))
-       (make-temp-file ".flymake_" nil (concat "_" basename))))))
+       (make-temp-file "flymake_" nil (concat "_" basename))))))
```

Comment:
What is the significance of this change?

Reply:
See: 
> I had to remove the leading dot from temp files, which appears to break relative imports. This will affect all checkers, but it could be made configurable per-checker

Python's module system can't handle file names with leading dots, so this causes some spurious errors when running mypy on temp files. I can provide an example, if you're interested. The `--shadow-file` flag avoids the issue but only works when the source file exists, so checking unsaved buffers would still be broken without this change.

Bit of an edge case, but making these short-lived temp files hidden didn't seem particularly important, either. Happy to revert or consider other options.

Reply:
Hmmm... the motivation is understandable but the end the result isn't ideal IMO :disappointed:. Dot prefixed files are a standard for hidden files on UNIX (and more recently windows since it stopped restricting their creation). Most systems (example file browsers) will auto hide them. Admittedly that isn't a big concern for something so short lived but I'd still like to retain the intent that these are hidden files, especially when their in a project directory instead of /tmp/.

Could you instead make this a generic parameter that can be set by the linter itself and keep the default as `.`. Should be just a new parameter to the define macro here (and the helpers below as well).

Reply:
Added a `temp-file-prefix` parameter with a default. After working with `flymake-collection-define` a bit, I'd have to agree with the user in #2 re: reducing the use of macros. Seems like a lot of the logic could be moved into functions in `flymake-collection.el`, which all the cherks already require.